### PR TITLE
Change contentContainer selector to more generic of new theme

### PR DESCRIPTION
**Pull request by ks129**
The current selector was made to fit specifically to the payment methods page, but with PrestaShop/PrestaShop#20737 carriers page is being migrated and this has to use more generic selector. This change generated HTML, but not how the page looks like.

### CODE REVIEW

**Review by atomiix — COMMENTED** [review]
CI should be fixed but this PR looks good

**Review by atomiix — APPROVED** [review]

**Comment by ks129** [comment]
@atomiix This PR doesn't touch PHP at all and PHPStan is what is failing, so this would be out-of-scope for this PR.

**Comment by atomiix** [comment]
@ks129 yep, that's why I approved it anyway :)

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

File: views/js/recommended-modules.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ var mbo = {};
     toolbarLastElement: '.toolbar-icons a:last-of-type',
     recommendedModulesButton: '#recommended-modules-button',
     oldButton: '#page-header-desc-configuration-modules-list',
-    contentContainer: '#main-div .content-div .container:last',
+    contentContainer: '#main-div .content-div .row:last .col',
```
Comment (NeOMakinG) [inline views/js/recommended-modules.js:43]: Hey 🖖 , thanks for your work!

What about adding new selectors, at the same time as keeping old selector ? Selecting with a generic class is worst to me as it's linked to the layout and can change anytime

Comment (ks129) [reply views/js/recommended-modules.js:43]: What about making this list of selectors and this use first found?

Comment (NeOMakinG) [reply views/js/recommended-modules.js:43]: What do you mean ? I think this is what I said, but differently 😄

Comment (ks129) [reply views/js/recommended-modules.js]: Is current solution better?

**Comment by ks129** [comment]
Note for reviewers: Code is duplicated because if I move selector selecting outside of done or fail, selector result is undefined.

File: views/js/recommended-modules.js [hunk 2]
```diff
@@ -40,7 +40,10 @@ var mbo = {};
     toolbarLastElement: '.toolbar-icons a:last-of-type',
     recommendedModulesButton: '#recommended-modules-button',
     oldButton: '#page-header-desc-configuration-modules-list',
-    contentContainer: '#main-div .content-div .container:last',
+    contentContainer: [
+      '#main-div .content-div .container:last',
+      '#main-div .content-div .col:last',
+    ],
```
Comment (NeOMakinG) [inline views/js/recommended-modules.js:46]:
```suggestion
    contentContainer: '#main-div .content-div .container:last, #main-div .content-div .js-content-col:last'
```

As it's temporary waiting for a new selector not related to the layout, and also I think we could just use a selector with `,` and select the first occurence of jQuery

File: views/js/recommended-modules.js [hunk 3]
```diff
@@ -142,7 +145,14 @@ var mbo = {};
...
 
-          $(pageMap.contentContainer).append(recommendedModulesContainer.getMarkup());
+          let contentContainer;
+          if (pageMap.contentContainer instanceof Array) {
+            contentContainer = pageMap.contentContainer.find(element => $(element).length);
+          } else {
+            contentContainer = pageMap.contentContainer;
+          }
+
+          $(contentContainer).append(recommendedModulesContainer.getMarkup());
```
Comment (NeOMakinG) [inline views/js/recommended-modules.js:155]:
```suggestion
          $(pageMap.contentContainer).first().append(recommendedModulesContainer.getMarkup());
```

So you could just do this with my previous comment

**Review by atomiix — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Comment by intraordinaire** [comment]
Sorry for the long delay on this PR.
The v2.x of ps_mbo will only receive security fixes now, so I close this one.
We are wokring on a total revamp of the module.

Anyway, thanks for your contribution :)

**Pull request closed without merging**